services/port_manager/src/main/java/com/futurewei/alcor/portmanager/restwrap/IpAddressRestWrap.java

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2019 The Alcor Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+        you may not use this file except in compliance with the License.
+        You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+        Unless required by applicable law or agreed to in writing, software
+        distributed under the License is distributed on an "AS IS" BASIS,
+        WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+        See the License for the specific language governing permissions and
+        limitations under the License.
+*/
+package com.futurewei.alcor.portmanager.restwrap;
+
+import com.futurewei.alcor.common.utils.Ipv4AddrUtil;
+import com.futurewei.alcor.common.utils.Ipv6AddrUtil;
+import com.futurewei.alcor.portmanager.exception.RangeIdNotFoundException;
+import com.futurewei.alcor.portmanager.exception.VerifySubnetException;
+import com.futurewei.alcor.portmanager.rollback.AbstractIpAddrRollback;
+import com.futurewei.alcor.portmanager.utils.BeanUtil;
+import com.futurewei.alcor.web.entity.ip.IpAddrRequest;
+import com.futurewei.alcor.web.entity.ip.IpVersion;
+import com.futurewei.alcor.web.entity.port.PortState;
+import com.futurewei.alcor.web.entity.SubnetStateJson;
+import com.futurewei.alcor.web.rest.IpAddressRest;
+import com.futurewei.alcor.portmanager.exception.IpAddrInvalidException;
+import com.futurewei.alcor.portmanager.exception.IpVersionInvalidException;
+import com.futurewei.alcor.portmanager.rollback.AllocateIpAddrRollback;
+import com.futurewei.alcor.portmanager.rollback.PortStateRollback;
+import com.futurewei.alcor.portmanager.rollback.ReleaseIpAddrRollback;
+import com.futurewei.alcor.web.rest.SubnetRest;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+
+public class IpAddressRestWrap {
```

Comment:
Maybe rename it to "IpAddressRestWrapper". It applies to other classes in the same directory and the directory itself.

Reply:
I always think wrap or wrapper is not a good name. Do you have any good suggestions?

Reply:
A few options: 
* IpAddressRestCoordinator
* IpAddressRestWorker
* IpAddressRestTask

@chenpiaoping It is up to you :-)